runimagejscript.py

```diff
@@ -0,0 +1,810 @@
+        " to your machine if cached dependencies not found."
+        init_type = self.init_choice.get_value()
+        if init_type != INIT_LOCAL:
+            # The component we attach the error to depends on if initialization has happened or not
+            if not imagej_process:
+                raise ValidationError(warn_msg, self.init_choice)
+            else:
+                raise ValidationError(warn_msg + " If re-initialization is required, please restart CellProfiler.",
+                                      self.initialized_method)
+
+
+    def run(self, workspace):
+        self.init_pyimagej()
+
+        if self.show_window:
+            workspace.display_data.script_input_pixels = {}
+            workspace.display_data.script_input_dimensions = {}
+            workspace.display_data.script_output_pixels = {}
+            workspace.display_data.script_output_dimensions = {}
+
+        script_filepath = path.join(self.script_directory.get_absolute_path(), self.script_file.value)
+        # convert the CP settings to script parameters for pyimagej
+        script_inputs = {}
+        for name in self.script_input_settings:
+            setting = self.script_input_settings[name]
+            if isinstance(setting, ImageSubscriber):
+                # Images need to be pulled from the workspace
+                script_inputs[name] = workspace.image_set.get_image(setting.get_value())
+                if self.show_window:
+                    workspace.display_data.script_input_pixels[name] = script_inputs[name].pixel_data
+                    workspace.display_data.script_input_dimensions[name] = script_inputs[name].dimensions
+            else:
+                # Other settings can be read directly
+                script_inputs[name] = setting.get_value()
+
+        # Start the script
+        to_imagej.put({PYIMAGEJ_KEY_COMMAND: PYIMAGEJ_CMD_SCRIPT_RUN, PYIMAGEJ_KEY_INPUT:
+            {PYIMAGEJ_SCRIPT_RUN_FILE_KEY: script_filepath,
+             PYIMAGEJ_SCRIPT_RUN_INPUT_KEY: script_inputs}
+                            })
+
+        # Retrieve script output
+        ij_return = from_imagej.get()
+        if ij_return != PYIMAGEJ_STATUS_CMD_UNKNOWN:
+            script_outputs = ij_return[PYIMAGEJ_KEY_OUTPUT]
+            for name in self.script_output_settings:
+                output_key = self.script_output_settings[name].get_value()
+                output_value = script_outputs[name]
+                output_image = Image(image=output_value, convert=False)
+                workspace.image_set.add(output_key, output_image)
+                if self.show_window:
+                    workspace.display_data.script_output_pixels[name] = output_image.pixel_data
+                    workspace.display_data.dimensions = output_image.dimensions
+        pass
+
+    def display(self, workspace, figure):
+        # TODO how do we handle differences in dimensionality between input/output images?
```

Comment:
Do you mean like, Z projecting so it goes from 3D to 2D? In general, I don't know that we want to support that, but worth a discussion.